src/platform_specific/affinity.unix.cc

```diff
@@ -0,0 +1,29 @@
+
+#include <cassert>
+#include <cstdint>
+
+#include <pthread.h>
+#include <sched.h>
+
+#include "affinity.h"
+
+namespace celerity {
+namespace detail {
+
+	uint32_t affinity_counter(const cpu_set_t& base_mask) {
+		auto count = 0;
+		for(auto i = 0; i < CPU_SETSIZE; i++) {
+			if(CPU_ISSET(i, &base_mask)) { ++count; }
+		}
+		return count;
+	}
+
+	uint32_t affinity_cores_available() {
+		cpu_set_t affinity_base_mask;
```

Comment:
Why is it called "base mask"? As far as I can tell this term was introduced as part of the original Insieme implementation and has no real meaning in this context.